### PR TITLE
Fix several tree2 transaction issues

### PR DESCRIPTION
**Pull request by Abe27342**
## Description

Fixes several issues related to transactions in tree2. This change is motivated by work to support transactions in optional-field, or at least anticipate places where they might necessitate changes to the persisted format. To that end, I started writing some axiomatic tests before realizing I wasn't totally sure how the production codepath worked with respect to ID allocation and several other concerns. Turns out it had some bugs at the branch level:

- AB#6484: Transactions starting on a point of divergence inadvertently included commits that they were rebased over. This resulted in various sorts of errors. This PR introduces a fix in `branch.ts` and adds some unit tests for the issue.
- AB#6485: The general rebase scheme does not preserve transactionality. In addition to the work item, I left some comments to this effect on the transaction-relevant private state in `Branch`. This PR does not attempt a fix for this issue.

The PR also adjusts the implementation for optional field's `invert`, which previously didn't support changes which might be made using transactions. Some tests for relevant scenarios are added to `editing.spec.ts` for that. This fixes some issues in previously skipped fuzz seeds.

### CODE REVIEW

**Comment by Abe27342** [comment]
Change is generally ready to review, but CI won't pass because:

1. Enabling transactions in fuzz tests also runs into the 0x428 bug that's fixed by #17829. I'd like to just merge that first to avoid having to disable a bunch of seeds.
2. A couple other fuzz tests fail with same errors as before. It's different seeds (RNG changed), but doesn't seem related to this PR.

File: packages/dds/tree/src/feature-libraries/optional-field/optionalField.ts
Comment (Abe27342) [inline packages/dds/tree/src/feature-libraries/optional-field/optionalField.ts:181]: Some axiomatic tests suggested that this might be better, though the choice is somewhat arbitrary (there were things not quite right with the test setup which convinced me to hold off checking them in to a later date, but this change seemed harmless enough to make now)

File: packages/dds/tree/src/shared-tree-core/branch.ts
Comment (taylorsw04) [inline packages/dds/tree/src/shared-tree-core/branch.ts:277]: could you move this logic (duplicated in abort) into pop?

File: packages/dds/tree/src/shared-tree-core/branch.ts
Comment (taylorsw04) [inline packages/dds/tree/src/shared-tree-core/branch.ts:150]: do you mean `rebaseOnto`?

**Comment by taylorsw04** [comment]
I'm happy with the changes to branch.ts and the tests; I think @yann-achard-MS should review the changes to optional field code.

File: packages/dds/tree/src/test/shared-tree/editing.spec.ts
Comment (yann-achard-MS) [inline packages/dds/tree/src/test/shared-tree/editing.spec.ts:2008]: Should that test also be inside the `describe("Transactions", ...`` block?

File: packages/dds/tree/src/test/shared-tree/editing.spec.ts
Comment (yann-achard-MS) [inline packages/dds/tree/src/test/shared-tree/editing.spec.ts:2217]: `tree` has had no edits applied to it. If that's by design then this line can go away

File: packages/dds/tree/src/test/shared-tree-core/branch.spec.ts
Comment (yann-achard-MS) [inline packages/dds/tree/src/test/shared-tree-core/branch.spec.ts:368]:
```suggestion
	describe("do not include rebased-over changes in a transaction", () => {
```

**Comment by yann-achard-MS** [comment]
> I'm happy with the changes to branch.ts and the tests; I think @yann-achard-MS should review the changes to optional field code.

@taylorsw04 - done. Looks like a strict improvement.

**Review by taylorsw04 — APPROVED** [review]

**Pull request merged**
